diff --git a/frontend/public/i18n.js b/frontend/public/i18n.js
index 733aa04dc6..737b44a99d 100644
--- a/frontend/public/i18n.js
+++ b/frontend/public/i18n.js
@@ -42,7 +42,7 @@ export const init = () => {
       lng: getLastLanguage(),
       fallbackLng: 'en',
       load: 'languageOnly',
-      debug: process.env.NODE_ENV === 'development',
+      debug: false,
       detection: { caches: [] },
       contextSeparator: '~',
       ns: [
@@ -97,9 +97,14 @@ export const init = () => {
       },
       saveMissing: true,
       missingKeyHandler: function(lng, ns, key) {
+        window.missingKeyNotified = window.missingKeyNotified ?? new Set();
+
         window.windowError = `Missing i18n key "${key}" in namespace "${ns}" and language "${lng}."`;
-        // eslint-disable-next-line no-console
-        console.error(window.windowError);
+        if (!window.missingKeyNotified.has(window.windowError)) {
+          window.missingKeyNotified.add(window.windowError);
+          // eslint-disable-next-line no-console
+          console.error(window.windowError);
+        }
       },
     })
     // Update loading promise and pass values and errors to the caller
